src/aap_eda/core/models/activation.py

```diff
@@ -74,6 +78,31 @@ class Meta:
     status_updated_at = models.DateTimeField(null=True)
     status_message = models.TextField(null=True, default=None)
 
+    def save(
+        self,
+        force_insert=False,
+        force_update=False,
+        using=None,
+        update_fields=None,
+    ):
+        # when creating
+        if update_fields is None and self.status == ActivationStatus.PENDING:
```

Comment:
Why are you adding pending status in this conditional? I think we want to check any transition and compare the current value with the old one. You can do this fetching the object from the DB before, but that adds an additional query to the DB, there is a way to do it without a query as is described here: https://stackoverflow.com/questions/23361057/django-comparing-old-and-new-field-value-before-saving/64116052#64116052

Reply:
We just update `status_message` whenever `status` is changed, no need to compare with the old value.